py/parsenum.c

```diff
@@ -218,7 +218,7 @@ mp_obj_t mp_parse_num_decimal(const char *str, size_t len, bool allow_imag, bool
         if (str + 2 < top && (str[1] | 0x20) == 'n' && (str[2] | 0x20) == 'f') {
             // inf
             str += 3;
-            dec_val = INFINITY;
+            dec_val = (mp_float_t)INFINITY;
```

Comment:
probably also use `MICROPY_FLOAT_CONST(x)` here

Reply:
This doesn't work since MICROPY_FLOAT_CONST either passes the value through untouched or adds an `F`. So if mp_float_t is double, we still get a compile error or if mp_float_t is float, then we get undefined `INFINITYF`.